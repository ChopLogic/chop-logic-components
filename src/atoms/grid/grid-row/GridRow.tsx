import React from 'react';
import { GridDataCell } from '@atoms/grid/data-cell/GridDataCell.tsx';
import { SelectGridRowCell } from '@atoms/grid/select-grid-row-cell/SelectGridRowCell.tsx';
import { getClassName } from '@utils';
import styles from './GridRow.module.scss';

type Props = {
  rowId: string;
  selectedIds: string[];
  selectRowById: (id: string) => void;
  deselectRowById: (id: string) => void;
  selectable: boolean;
  values: (string | React.ReactElement)[];
  disabled?: boolean;
};

export const GridRow: React.FC<Props> = ({ rowId, selectRowById, deselectRowById, selectable, values, selectedIds, disabled }) => {
  const selected = selectedIds.includes(rowId);

  return (
    <tr className={getClassName([styles.row, { [styles.row__selected]: selected }])}>
      {selectable && (
        <SelectGridRowCell
          rowId={rowId}
          selectRowById={selectRowById}
          deselectRowById={deselectRowById}
          isRowSelected={selected}
          disabled={disabled}
        />
      )}
      {values.map((value, index) => (
        <GridDataCell key={`row_${index}`} value={value} />
      ))}
    </tr>
  );
};
