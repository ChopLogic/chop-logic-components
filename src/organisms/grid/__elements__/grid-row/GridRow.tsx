import { getClassName } from '@utils';
import React from 'react';

import { GridDataCell } from '../data-cell/GridDataCell';
import { SelectGridRowCell } from '../select-grid-row-cell/SelectGridRowCell';
import styles from './GridRow.module.scss';

type Props = {
  rowId: string;
  selectedIds: string[];
  selectRowById: (id: string) => void;
  deselectRowById: (id: string) => void;
  selectable: boolean;
  values: (string | React.ReactElement)[];
  disabled?: boolean;
};

export const GridRow: React.FC<Props> = ({ rowId, selectRowById, deselectRowById, selectable, values, selectedIds, disabled }) => {
  const selected = selectedIds.includes(rowId);

  return (
    <tr className={getClassName([styles.row, { [styles.row__selected]: selected }])}>
      {selectable && (
        <SelectGridRowCell
          rowId={rowId}
          selectRowById={selectRowById}
          deselectRowById={deselectRowById}
          isRowSelected={selected}
          disabled={disabled}
        />
      )}
      {values.map((value, index) => (
        <GridDataCell key={`row_${index}`} value={value} />
      ))}
    </tr>
  );
};
