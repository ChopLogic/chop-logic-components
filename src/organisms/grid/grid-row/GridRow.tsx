import { getClassName } from '@utils';
import { FC, ReactElement } from 'react';

import { GridDataCell } from '../data-cell/GridDataCell.tsx';
import { SelectGridRowCell } from '../select-grid-row-cell/SelectGridRowCell.tsx';
import styles from './GridRow.module.scss';

type Props = {
  rowId: string;
  selectedIds: string[];
  selectRowById: (id: string) => void;
  deselectRowById: (id: string) => void;
  selectable: boolean;
  values: (string | ReactElement)[];
  disabled?: boolean;
};

export const GridRow: FC<Props> = ({ rowId, selectRowById, deselectRowById, selectable, values, selectedIds, disabled }) => {
  const selected = selectedIds.includes(rowId);

  return (
    <tr className={getClassName([styles.row, { [styles.row__selected]: selected }])}>
      {selectable && (
        <SelectGridRowCell
          rowId={rowId}
          selectRowById={selectRowById}
          deselectRowById={deselectRowById}
          isRowSelected={selected}
          disabled={disabled}
        />
      )}
      {values.map((value, index) => (
        <GridDataCell key={`row_${index}`} value={value} />
      ))}
    </tr>
  );
};
