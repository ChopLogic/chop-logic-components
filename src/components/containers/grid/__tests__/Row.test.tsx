import { render, screen } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';

import GridRow, { GridRowProps } from '../elements/Row';

describe('GridRow', () => {
  const testProps: GridRowProps = {
    rowId: 'item-1',
    selectRowById: vi.fn(),
    deselectRowById: vi.fn(),
    selectable: true,
    values: ['Value 1', 'Value 2', <span key='value-3'>Value 3</span>],
    selectedIds: ['value1', 'value2', 'value3'],
  };

  it('should match the snapshot', () => {
    const { asFragment } = render(
      <table>
        <tbody>
          <GridRow {...testProps} />
        </tbody>
      </table>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should not render a checkbox if the row is not selectable', () => {
    render(
      <table>
        <tbody>
          <GridRow {...testProps} selectable={false} />
        </tbody>
      </table>,
    );
    expect(screen.queryByRole('checkbox')).not.toBeInTheDocument();
  });
});
