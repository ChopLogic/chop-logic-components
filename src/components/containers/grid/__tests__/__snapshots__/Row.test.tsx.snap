// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GridRow > should match the snapshot 1`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr
        class="sc-kpOvIu fjRctD"
      >
        <td
          class="sc-iQQCXo jmwnoV"
        >
          <div
            class="sc-jMsorb dtJOel"
          >
            <input
              class="sc-cXawGu iLZNZv"
              id="select_row_item-1"
              type="checkbox"
            />
            <label
              class="sc-jScdur laKqIp"
              for="select_row_item-1"
            >
              <svg
                fill="currentColor"
                height="1em"
                role="img"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M14 0H2C.9 0 0 .9 0 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V2c0-1.1-.9-2-2-2zm0 14H2V2h12v12z"
                  fill="currentColor"
                />
              </svg>
            </label>
          </div>
        </td>
        <td
          class="sc-iQQCXo jmwnoV"
        >
          Value 1
        </td>
        <td
          class="sc-iQQCXo jmwnoV"
        >
          Value 2
        </td>
        <td
          class="sc-iQQCXo jmwnoV"
        >
          <span>
            Value 3
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
